src/igcontrolbase/igcontrolbase.ts

```diff
@@ -230,6 +230,13 @@ export class IgControlBase<Model> implements DoCheck {
 		if (o1 === o2) { return true; }
 		if (o1 === null || o2 === null) { return false; }
 		if (o1 !== o1 && o2 !== o2) { return true; }// NaN === NaN
+		var isDOMnode = typeof Node === "object" ? o1 instanceof Node : 
```

Comment:
@tbelcheva Please put all the code in a separate function(s) and also add tests for them.